Archipelago.HollowKnight/IC/ArchipelagoUIDef.cs

```diff
@@ -5,17 +5,31 @@ namespace Archipelago.HollowKnight.IC
 {
     internal class ArchipelagoUIDef : MsgUIDef
     {
-        public ArchipelagoUIDef(UIDef def)
+        public ArchipelagoUIDef(UIDef def, string targetSlotName = null)
         {
             if (def is MsgUIDef msgDef)
             {
-                name = msgDef.name.Clone();
+                if (targetSlotName == null)
+                {
+                    name = msgDef.name.Clone();
+                }
+                else
+                {
+                    name = new BoxedString($"{targetSlotName}'s {msgDef.name.Value}");
```

Comment:
This should probably use GetPreviewName() rather than msgDeg.name.Value. This is just in case the implementation of MsgUIDef changes in the future. (Or if IString changes, we can let MsgUIDef tell us the name since if IString changes then MsgUIDef will change to suit.)

Reply:
Fixed in latest commit.